### ISSUE TITLE
Selection tool makes entire app transparent

### ISSUE THREAD

**Issue by mquevill** [OpenChemistry/avogadrolibs#653]
**Avogadro version: 1.94.0-17-gb2b873b**
 - Avogadrolibs: 1.94.0-76-g5c867a2
 - Qt: 5.12.8

**Desktop version:**
 - OS & Version: Linux Mint 20.1 (based on Ubuntu 20.04)
 - Compiler: gcc 9.3.0

**Describe the bug**
The selection tool makes the entire app transparent, such that you can see whatever is behind it. This happens when selecting atoms or when dragging a box selection. This was originally raised in OpenChemistry/avogadroapp#167 in a comment (see below).

**To Reproduce**
Select an atom or create a box with the selection tool.

**Screenshots**
Just one atom selected:
![image](https://user-images.githubusercontent.com/17033543/124786927-1f293180-df16-11eb-8290-b59ee5ddc90b.png)

Box selection tool:
![Screenshot from 2021-07-07 11-26-42](https://user-images.githubusercontent.com/17033543/124787195-526bc080-df16-11eb-8bc6-e24c211e7311.png)

# Original comment:
There's a separate transparency issue with the selection tool. With version 1.9.4, when a box is drawn using the cursor to select some atoms, the box has some transparency so any windows underneath the avogadro app become visible.  Likewise the pale blue halo around selected atoms exposes the underlying windows.

This problem is less black & white than the background colour issue - the selection halo should be transparent within the app! But it shouldn't expose windows outside the app.

I haven't been able to identify what's controlling it. There's nothing immediately obvious like there was for the backgroundColor setting. The only  transparency control I can see in avogadrolibs/avogadro/qtplugins/selectiontools is [`mesh->setOpacity(180)`](https://github.com/OpenChemistry/avogadrolibs/blob/5c867a2478af0b8e535fffd3f82838cea2dc4b3e/avogadro/qtplugins/selectiontool/selectiontool.cpp#L220).  There's also[ `selectedSpheres->setOpacity(0.42)`](https://github.com/OpenChemistry/avogadrolibs/blob/5c867a2478af0b8e535fffd3f82838cea2dc4b3e/avogadro/qtplugins/ballandstick/ballandstick.cpp#L64) in qtplugins/ballandstick.

_Originally posted by @drew-parsons in https://github.com/OpenChemistry/avogadroapp/issues/167#issuecomment-874691372_

**Comment by ghutchis** [comment]
I'm not sure if it's the exact same bug, but a Qt bug report seemed to suggest this as a strategy.

**Comment by drew-parsons** [comment]
That looks like the right fix.  I'm not sure that nvidia was the problem (it's actually the other way around, it was already working fine with nvidia), but the glBlendFuncSeparate patch resolves the transparency problem all the same.

I've got a hybrid system (bumblebee/primus) with intel or nvidia video.  I could reproduce the original transparency problem using the intel driver. When I switched to the nvidia driver (`primusrun avogadro2`), the problem did not appear. The selection square and the selected atoms did not expose the underlying windows when running with nvidia.

The PR #656 patch applying glBlendFuncSeparate fixes the problem for the intel driver.  And the nvidia driver continues to run avogadro2 without problems.

PR #656 looks like a good solution to me.
